src/App/PropertyStandard.cpp

```diff
@@ -2827,10 +2835,21 @@ void PropertyMaterialList::setDiffuseColor(int index, uint32_t rgba)
     hasSetValue();
 }
 
+void PropertyMaterialList::setDiffuseColors(const std::vector<App::Color>& colors)
+{
+    aboutToSetValue();
+    setSize(colors.size(), _lValueList[0]);
+
+    for (int i = 0; i < colors.size(); i++) {

```

Comment:
Any reason not to use range-for here?

Reply:
Yes. We're indexing two different arrays